src/app/js/formats/streamgraph/Streamgraph.js

```diff
@@ -0,0 +1,410 @@
+            // Get the hovered values for tooltip
+            // set hover and display tooltip
+            // *******************************************************
+
+            let mousex = d3.mouse(this)[0];
+            let date = componentContext.xAxisScale.invert(mousex);
+
+            hoveredKey = d3.select(this).attr("name");
+
+            for (let elem of d) {
+                if (elem.data.date.getFullYear() == date.getFullYear()) {
+                    hoveredValue = elem.data[hoveredKey];
+                    break;
+                }
+            }
+
+            d3.select(this)
+                .classed("hover", true)
+                .attr("stroke", "#000")
+                .attr("stroke-width", "0.5px"),
+                tooltip.html("<p>" + hoveredKey + "<br>" + hoveredValue + "</p>").style("visibility", "visible");
+        }).on("mouseover", (d, i) => {
+
+            // *******************************************************
+            // set the hover opacity
+            // *******************************************************
+
+            mouseIsOverStream = true;
+            this.streams.transition()
+                .duration(25)
+                .attr("opacity", (d, j) => {
+                    return j != i ? 0.3 : 1;
+                })
+        }).on("mouseout", function (d, i) {
+
+            // *******************************************************
+            // unset hover and hide tooltip
+            // *******************************************************
+
+            mouseIsOverStream = false;
+            componentContext.streams.transition()
+                .duration(25)
+                .attr("opacity", "1");
+            d3.select(this)
+                .classed("hover", false)
+                .attr("stroke-width", "0px"),
+                tooltip.html("<p>" + hoveredKey + "<br>" + hoveredValue + "</p>").style("visibility", "hidden");
+        });
+    }
+
+    componentDidMount() {
+        this.setGraph();
+    }
+
+    componentWillUpdate() {
+        d3.selectAll("#d3DivContainer" + this.uniqueId).selectAll("div").remove();
+        d3.selectAll("#d3DivContainer" + this.uniqueId).remove();
+        d3.selectAll("#" + this.anchor).selectAll("g").remove();
+        d3.selectAll("#" + this.anchor).selectAll("defs").remove();
+    }
+
+    componentDidUpdate() {
+        this.setGraph();
+    }
+
+    render() {
+        const { width, height } = this.state;
+        return (
+            <div id={this.divContainer} ref="divContainer" style={styles.divContainer}>
+                <svg id={this.svgContainer} ref="svgContainer" width={width} height={height}>
+                    <g id={this.anchor} ref="anchor" />
+                </svg>
+            </div>
+        );
+    }
+};
+
+const mapStateToProps = (state, { formatData }) => {
+    //console.log("mapStateToProps - state:");
+    //console.log(state);
+    return ({});
```

Comment:
If you `mapStateToProps` returns nothing, you don't have to write it.
Please delete this function and remove the `connect` below.